clients/java/signalr/src/main/java/JsonHubProtocol.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.util.ArrayList;
+
+public class JsonHubProtocol implements IHubProtocol{
+    private JsonParser jsonParser = new JsonParser();
+    private static final String RECORD_SEPARATOR = "\u001e";
+
+    @Override
+    public String name() {
+        return "json";
+    }
+
+    @Override
+    public int version() {
+        return 1;
```

Comment:
Wait... You're just returning a value you need instead of creating a nested static factory class that returns a VersionManager that returns a value of a private a final static UPPER_CASED variable containing the version (or an enum)? Dude, your break from Java was too long...

Reply:
Needs moar `JsonHubProtocolVersionIdentifierFactory`